tests/capture.rs

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+use std::net::{IpAddr, Ipv4Addr, SocketAddr};
+
+use tokio::time::{timeout, Duration};
+
+use quilkin::{
+    config::{Builder, Filter},
+    endpoint::Endpoint,
+    filters::{capture, token_router},
+    metadata::MetadataView,
+    test_utils::TestHelper,
+};
+
+/// This test covers both token_router and capture filters,
+/// since they work in concert together.
+#[tokio::test]
+async fn token_router() {
+    let mut t = TestHelper::default();
+    let echo = t.run_echo_server().await;
+
+    let capture_yaml = r#"
+regex:
+    pattern: .{3}$
```

Comment:
I actually didn't realise that you could do neat stuff like this! 😮 (I always relate regex with doing character matching) -- this is very, very cool, I hadn't thought through the implications of this addition.

I think it would be awesome to show in capture.md under "Configuration Examples" that you could do first 3 values capturing using `prefix`, but also do the same thing using `regex` -- two different ways to do the same thing, just to give people ideas of what is possible.

Very nice.